### DISCUSSION TITLE
Errors handlers configuration

### DISCUSSION THREAD

**Discussion by alan994** [ory/oathkeeper#960] [Q&A] [answered]
Hi Ory team,

I have a question regarding error handlers in the `OathKeeper` project. I'm trying to configure a redirect error handler if my incoming request is unauthorized. The logic is simple if a request contains the `Accept: text/*` header, use the `redirect` handler otherwise use the `JSON` handler. 

My OathKeeper config:
```
...
errors:  
  fallback:
    - json

  handlers:
    redirect:
      enabled: true
      config:
        to: https://www.google.com
        when:
          - error:
              - unauthorized
              - forbidden
            request:
              header:
                accept:
                  - text/html
    json:
      enabled: true
      config:
        verbose: true
...
```
Access rule config:
```
...
- id: "alan:api:cookie2"
  upstream:
    preserve_host: false # this will ensure we use host and port of `url` property bellow
    url: "http://api" #5291
    strip_path: "/api/cookie"
  match:
    url: "http://127.0.0.1:4456/api/cookie/secure"
    methods:
      - GET
  authenticators:
    - handler: cookie_session
  authorizer:
    handler: allow
  mutators:
    - handler: hydrator
  error:
    - handler: redirect
...
```

I would expect that if I try to call `http://127.0.0.1:4456/decisions/api/cookie/secure` from my browser I would be redirected to `https://www.google.com` but instead, I get a `JSON` error like this:
```
{
  "error": {
    "code": 401,
    "status": "Unauthorized",
    "message": "Access credentials are invalid"
  }
}
```

The interesting part is if I change the `fallback` property to `redirect` in my `OathKeeper` configuration it works as expected. I'm redirected to google.

I have a code sample, you can find it [here](https://github.com/alan994/ory-playground)

Any help or suggestion much appreciated.

**Comment by alan994** [comment, marked as answer]
@aeneasr @gen1us2k 

Maybee some ideas about what I'm doing wrong?

**Reply by vinckr** [reply]
Hello @alan994 

apologies for the very late answer 😅, just checking in if your problem still persists. 

We also have a hydrator mutator example here now: https://github.com/ory/examples/tree/master/oathkeeper/04-hydrator-mutator

I will mark this as answered for now, but feel free to come back to the discussion any time.
Thanks a ton :bee:
